scalableminds/webknossos
# Don't remove other volume layers when opening volume annotation

When creating a new annotation with a volume layer (without fallback) for a dataset which has an existing segmentation layer:
- [X] keep on-disk segmentation layers in the annotation so that they can be activated
- [X] hide on-disk segmentation layer on first view and make new volume layer visible
- [x] ensure that changing the layer visibility of a layer in an annotation, does not change the layer visibility for the dataset settings. otherwise, opening the dataset would likely show no segmentation layer.

### URL of deployed dev instance (used for testing):
- https://keepseglayers.webknossos.xyz

### Steps to test:
- Open a dataset with a segmentation layer (keep that dataset in an own tab)
- Hide the segmentation layer & refresh -> layer should still be hidden
- Unhide the segmentation layer & refresh -> layer should still be shown
- Open a second tab to create an annotation for that dataset via the dashboard (use the advanced menu) and select that a new segmentation should be created
- The empty segmentation layer should be visible by default. Also, the original segmentation layer should still be listed in the left sidebar (but hidden by default).
- Reload the annotation -> visibilities shouldn't have changed (i.e., the original segmentation layer should be invisible)
- Reload the dataset in tab 1 -> visibilities shouldn't have changed there (i.e., the original segmentation layer should still be visible).
- Make the original segmentation layer visible in tab 2 and reload --> original segmentation layer should still be visible.

### Issues:
- fixes #6143

------
(Please delete unneeded items, merge only when none are left open)
